extensions/WSLExtension/DevHomeProviders/WslProvider.cs

```diff
@@ -97,4 +109,31 @@ public IAsyncOperation<ComputeSystemsResult> GetComputeSystemsAsync(IDeveloperId
         var notImplementedException = new NotImplementedException($"Method not implemented by WSL Compute System Provider");
         return new ComputeSystemAdaptiveCardResult(notImplementedException, notImplementedException.Message, notImplementedException.Message);
     }
+
+    /// <summary>
+    /// Gets the localized error message for when the virtual machine platform Windows optional component
+    /// is not enabled.
+    /// </summary>
+    /// <returns>
+    /// Returns the error message for when the feature is not enabled and returns an empty string when the feature is enabled.
+    /// </returns>
+    private string GetErrorStringForWhenFeatureNotEnabled()
+    {
+        var searcher = new ManagementObjectSearcher($"SELECT InstallState FROM Win32_OptionalFeature WHERE Name = 'VirtualMachinePlatform'");
+        var collection = searcher.Get();
+
+        foreach (var instance in collection)
+        {
+            var enablementState = instance?.GetPropertyValue("InstallState") as uint?;
+            _log.Information($"Found VirtualMachinePlatform feature with enablement state: '{enablementState}'");
+
+            if (enablementState != FeatureEnabledState)

```

Comment:
Isn't enablementState a `uint?` what if enablmentState is null here?

Reply:
thats fine in this case as it means its not the enabled state we're looking for. There is nothing we can do in this case. we treat it the same as if it was disabled.